app/(app)/courses/[slug]/[id]/_client.tsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+import { FEATURE_FLAGS, isFlagEnabled } from "@/utils/flags";
+import { CircleCheck, SquarePlay } from "lucide-react";
+import { type Session } from "next-auth";
+import { notFound } from "next/navigation";
+import { mockContentList } from "../../mock";
+
+interface ContentProps {
+  session: Session | null;
+}
+
+const Content = ({ session }: ContentProps) => {
+  const flagEnabled = isFlagEnabled(FEATURE_FLAGS.COURSE_VIDEO);
+
+  if (!flagEnabled) {
+    notFound();
+  }
```

Comment:
Possible stupid question on this one Carolina. Can you put featureflag the route? Then you can just let nexts router deal with the notFound ?

Reason this might be a stupid queston is I have no idea how nexts router works 🤣  I am basing this comment from my knowledge of react router

Reply:
I'm not sure how we could but that's a good pattern that we could have. We will figure it out